security/authorization/compiler/api/compiler_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package compiler
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"testing"
+
+	pb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	"google.golang.org/protobuf/proto"
+)
+
+var testAction = "ALLOW"
+var testPolicies = make(map[string]string)
+
+func TestParseEval(t *testing.T) {
```

Comment:
Could you add more test on `compileCel()`?

E.g., create test policy that have some attributes that are not defined, we will expect to see failure, such as
`connection.peer_certificate`, `response.size`.

Reply:
I added a simple loop to confirm that invalid attribute names will fail. I manually created the test policy to avoid having to create a bad_example_policy.yaml that we need to parse and feed when compileCel() only worries about the direct given expr string